### ISSUE TITLE
Path formatter breaks on inheritance

### ISSUE THREAD

**Issue by NiklasBr** [pimcore/pimcore#9509]
# Bug Report

### Expected behavior

This is what the root object looks like:
![image](https://user-images.githubusercontent.com/279826/122745882-c016bc00-d289-11eb-84bc-0ff442014902.png)

Inherited values, like their parent values, should be displayed when using [Path Formatter](https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Class_Settings/Path_Formatter.html), but are not.

### Actual behavior

The content becomes blank, like the inherited values here:

![image](https://user-images.githubusercontent.com/279826/122746119-fc4a1c80-d289-11eb-94df-8508df5ad4cc.png)

### Steps to reproduce

See code example on a `Product` class with variants.

### Code example
```php
<?php

namespace AppBundle\Service;

use AppBundle\Model\DataObject\Product;
use Pimcore\Model\DataObject\ClassDefinition\PathFormatterInterface;
use Pimcore\Model\DataObject\Concrete;
use Pimcore\Model\Element\ElementInterface;

class PathFormatterService implements PathFormatterInterface
{
    public function formatPath(array $result, ElementInterface $source, array $targets, array $params): array
    {
        foreach ($targets as $key => $item) {
            if ($item["type"] === "object") {
                $targetObject = Concrete::getById($item["id"]);
                if ($targetObject instanceof Product) {
                    $result[$key] = $this->getPath($targetObject, 'getOSName');
                }
            }
        }

        return $result;
    }

    protected function getPath(Concrete $object, string $getter): string
    {
        $result = [];

        while ($object instanceof Concrete) {
            if ($value = $object->{$getter}()) {
                $result[] = $value;
            }

            $object = $object->getNextParentForInheritance();
        }

        return implode(' &rarr; ', array_reverse($result));
    }
}
```

**Comment by NiklasBr** [comment]
The error appears to happen here: 

https://github.com/pimcore/pimcore/blob/de89d156b4212cc6d408d9d1729c07e13ac90653/bundles/AdminBundle/Controller/Admin/ElementController.php#L567-L584

Because `$editModeData` is an empty array it never gets replaced with `$result`

**Comment by brusch** [comment]
Fixed by #9524
